src/test/scala/org/kibanaLoadTest/helpers/ResponseParser.scala

```diff
@@ -0,0 +1,130 @@
+        // Request:
+        strLine = br.readLine.trim
+        // login: OK
+        val status = STATUS_REGEXP.findFirstIn(strLine).getOrElse("")
+        val name =
+          strLine
+            .replaceAll(STATUS_MESSAGE_REGEXP, "")
+            .trim
+            .replaceAll(".$", "");
+        br.readLine
+        // =========================
+        strLine = br.readLine
+        // Session:
+        var sessionValue = ""
+        strLine = br.readLine
+        while (strLine != SPLIT_LINE) {
+          sessionValue = sessionValue + strLine.trim
+          strLine = br.readLine
+        }
+        val userId = sessionValue.split(",")(1)
+        strLine = br.readLine
+        // HTTP request:
+        val requestStr = br.readLine()
+        // POST http://localhost:5620/internal/security/login
+        val methodAndUrl = requestStr.split(" ", 2)
+        val method = methodAndUrl(0)
+        val url = if (methodAndUrl.length == 2) methodAndUrl(1) else ""
+        strLine = br.readLine
+        // headers:
+        strLine = br.readLine
+        val requestHeaders = getHeaders(SPLIT_LINE)
+        val requestBody =
+          REQUEST_BODY_REGEXP
+            .findFirstIn(getBodyString(SPLIT_LINE))
+            .getOrElse("")
+        // =========================
+        br.readLine()
+        // HTTP response:
+        strLine = br.readLine()
+        // status:
+        val responseStatus =
+          RESPONSE_STATUS_CODE_REGEXP.findFirstIn(br.readLine()).getOrElse("")
+        strLine = br.readLine()
+        //headers:
+        strLine = br.readLine()
+        val responseHeaders =
+          getHeaders(RECORD_END_LINE)
+        val responseBody = RESPONSE_BODY_REGEXP
+          .findFirstIn(getBodyString(RECORD_END_LINE))
+          .getOrElse("")
+
+        responseList += Response(
+          userId,
+          name,
+          status,
+          method,
+          url,
+          requestHeaders,
+          requestBody,
+          responseStatus,
+          responseHeaders,
+          responseBody,
+          0L,
+          0L,
+          ""
+        )
+      }
+      strLine = br.readLine
+      // <<<<<<<<<<<<<<<<<<<<<<<<<
+    }
+    fsStream.close()
+
+    responseList
+  }
+
+  private def getBodyString(endLine: String): String = {
+    var bodyString = ""
+    if (strLine.startsWith(BODY_LINE)) {
+      while (!strLine.startsWith(endLine)) {
+        bodyString += strLine.trim
+        strLine = br.readLine()
+      }
+    }
+
+    // remove multiple spaces
+    bodyString.replaceAll("\\s+", " ")
+  }
+
+  private def getHeaders(endLine: String): String = {
+    val gson = new Gson
+    val jsonObject = new JsonObject()
+    while (!strLine.startsWith(BODY_LINE) && !strLine.startsWith(endLine)) {
+      val values = strLine.trim.split(":", 2)
+      if (values.length == 2) {
+        jsonObject.addProperty(values(0).trim, values(1).trim)
+      }
+      strLine = br.readLine
+    }
+    gson.toJson(jsonObject)
```

Comment:
Ever possible for this to throw an exception?  Not sure.

Reply:
At least not here, cause I'm parsing manually added JsonObject and in worse case it will be empty json... That what I think :)